### PR TITLE
[WIP]Pipeline: Support pipeline execution summary

### PR DESCRIPTION
**Pull request by ywqzzy**
### What problem does this PR solve?

Issue Number: ref #6518 

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Flash/Coprocessor/DAGContext.h [hunk 1]
```diff
@@ -328,14 +341,22 @@ class DAGContext
...
     /// join_execute_info_map is a map that maps from join_probe_executor_id to JoinExecuteInfo
     /// DAGResponseWriter / JoinStatistics gets JoinExecuteInfo through it.
     std::unordered_map<std::string, JoinExecuteInfo> join_execute_info_map;
-    /// profile_streams_map is a map that maps from executor_id (table_scan / exchange_receiver) to BlockInputStreams.
+    /// inbound_io_input_streams_map is a map that maps from executor_id (table_scan / exchange_receiver) to BlockInputStreams.
     /// BlockInputStreams contains ExchangeReceiverInputStream, CoprocessorBlockInputStream and local_read_input_stream etc.
     std::unordered_map<String, BlockInputStreams> inbound_io_input_streams_map;
+
+    /// io_source_ops_map is a map that maps from executor_id (table_scan / exchange_receiver) to SourceOps.
+    std::unordered_map<String, SourceOps> io_source_ops_map;
```
Comment (SeaRise) [inline dbms/src/Flash/Coprocessor/DAGContext.h:358]: how about just holding the executor_id of `io_input/io_source`, and then we don't need to hold two maps `inbound_io_input_streams_map` and `io_source_ops_map` here.

File: dbms/src/Operators/Operator.h [hunk 2]
```diff
@@ -101,9 +108,8 @@ class SourceOp : public Operator
 
     OperatorStatus awaitImpl() override { return OperatorStatus::HAS_OUTPUT; }
 };
-using SourceOpPtr = std::unique_ptr<SourceOp>;
+using SourceOpPtr = std::shared_ptr<SourceOp>;
```
Comment (SeaRise) [inline dbms/src/Operators/Operator.h:111]: looks a bit hacky.
How about just holding the executor_id in `io_sources_map` and holding the connection_info in `DAGContext`?
`Map<String, ConnectionInfos> receiver_connection_infos_map`.

File: dbms/src/Flash/Pipeline/Schedule/Events/Event.h [hunk 3]
```diff
@@ -86,6 +89,8 @@ class Event : public std::enable_shared_from_this<Event>
 
     LoggerPtr log;
 
+    Stopwatch stop_watch;
```
Comment (SeaRise) [inline dbms/src/Flash/Pipeline/Schedule/Events/Event.h:92]: Seems that just holding the stop_watch in `PipelineEvent` is ok.

File: dbms/src/Flash/Pipeline/Schedule/Events/PipelineEvent.h [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        Context & context_,
+        size_t concurrency_)
+        : Event(exec_status_, std::move(mem_tracker_), req_id)
+        , context(context_)
+        , concurrency(concurrency_)
+    {}
+
+protected:
+    Context & context [[maybe_unused]];
+    size_t concurrency [[maybe_unused]];
```
Comment (SeaRise) [inline dbms/src/Flash/Pipeline/Schedule/Events/PipelineEvent.h:39]: Seems all useless here.

File: dbms/src/Flash/Planner/Plans/PhysicalFilter.cpp [hunk 5]
```diff
@@ -61,13 +61,16 @@ void PhysicalFilter::buildBlockInputStreamImpl(DAGPipeline & pipeline, Context &
 void PhysicalFilter::buildPipelineExecGroup(
     PipelineExecutorStatus & exec_status,
     PipelineExecGroupBuilder & group_builder,
-    Context & /*context*/,
+    Context & context,
     size_t /*concurrency*/)
 {
+    auto & executor_profile = context.getDAGContext()->getPipelineProfilesMap()[executor_id];
```
Comment (SeaRise) [inline dbms/src/Flash/Planner/Plans/PhysicalFilter.cpp:67]: A data race may occur when an entry is inserted into a map

File: dbms/src/Flash/Pipeline/Schedule/Events/PipelineEvent.h (hunk 4 above)
Comment (ywqzzy) [reply dbms/src/Flash/Pipeline/Schedule/Events/PipelineEvent.h:39]: Okay, i will remove it.

**Pull request closed without merging**